RoundsWithFriends/GameModes/RWFGameMode.cs

```diff
@@ -0,0 +1,462 @@
+            this.teamRounds.Clear();
+            foreach (Player player in PlayerManager.instance.players)
+            {
+                this.PlayerJoined(player);
+            }
+
+            GameManager.instance.isPlaying = true;
+            this.StartCoroutine(this.DoStartGame());
+        }
+
+        protected virtual IEnumerator DoStartGame()
+        {
+            CardBarHandler.instance.Rebuild();
+            UIHandler.instance.InvokeMethod("SetNumberOfRounds", (int) GameModeManager.CurrentHandler.Settings["roundsToWinGame"]);
+            ArtHandler.instance.NextArt();
+
+            yield return GameModeManager.TriggerHook(GameModeHooks.HookGameStart);
+
+            GameManager.instance.battleOngoing = false;
+
+            UIHandler.instance.ShowJoinGameText("LETS GOO!", PlayerSkinBank.GetPlayerSkinColors(1).winText);
+            yield return new WaitForSeconds(0.25f);
+            UIHandler.instance.HideJoinGameText();
+
+            PlayerManager.instance.SetPlayersSimulated(false);
+            PlayerManager.instance.InvokeMethod("SetPlayersVisible", false);
+            MapManager.instance.LoadNextLevel(false, false);
+            TimeHandler.instance.DoSpeedUp();
+
+            yield return new WaitForSecondsRealtime(1f);
+
+            yield return GameModeManager.TriggerHook(GameModeHooks.HookPickStart);
+
+            for (int i = 0; i < PlayerManager.instance.players.Count; i++)
+            {
+                yield return this.WaitForSyncUp();
+
+                yield return GameModeManager.TriggerHook(GameModeHooks.HookPlayerPickStart);
+
+                CardChoiceVisuals.instance.Show(i, true);
+                yield return CardChoice.instance.DoPick(1, PlayerManager.instance.players[i].playerID, PickerType.Player);
+
+                yield return GameModeManager.TriggerHook(GameModeHooks.HookPlayerPickEnd);
+
+                yield return new WaitForSecondsRealtime(0.1f);
+            }
+
+            yield return this.WaitForSyncUp();
+            CardChoiceVisuals.instance.Hide();
+
+            yield return GameModeManager.TriggerHook(GameModeHooks.HookPickEnd);
+
+            MapManager.instance.CallInNewMapAndMovePlayers(MapManager.instance.currentLevelID);
+            TimeHandler.instance.DoSpeedUp();
+            TimeHandler.instance.StartGame();
+            GameManager.instance.battleOngoing = true;
+            UIHandler.instance.ShowRoundCounterSmall(this.teamPoints, this.teamRounds);
+            PlayerManager.instance.InvokeMethod("SetPlayersVisible", true);
+
+            this.StartCoroutine(this.DoRoundStart());
+        }
+
+        protected virtual IEnumerator RoundTransition(int winningTeamID)
+        {
+            yield return GameModeManager.TriggerHook(GameModeHooks.HookPointEnd);
+            yield return GameModeManager.TriggerHook(GameModeHooks.HookRoundEnd);
+
+            if (this.teamRounds[winningTeamID] >= (int) GameModeManager.CurrentHandler.Settings["roundsToWinGame"])
```

Comment:
A win condition is such a universal requirement for gamemodes that I think moving it to gamemode handlers makes more sense than checking some arbitrary setting here. Something like `bool IsGameOver();` or `int GetGameWinner()` (-1 if no winner) ?

Reply:
I agree. I think I like `int GetGameWinner() -> teamID` the most.